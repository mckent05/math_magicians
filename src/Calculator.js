import React from 'react'
import './App.css'

Calculator = () => {
    return (
        <div className='btn-cont'>
            <div className='result d-flex a-center'>
                <p>0</p>
            </div>
            <div className='btn-container'>
                <div className='first-row'>
                    <button className='button' data-id='AC'>AC</button>
                    <button className='button' data-id='oper'>+/-</button>
                    <button className='button' data-id='oper'>%</button>
                    <button className='button' data-id='oper'> / </button>
                </div>
                <div className='second-row'>
                    <button className='button' data-id='number'>7</button>
                    <button className='button' data-id='number'>8</button>
                    <button className='button' data-id='number'>9</button>
                    <button className='button' data-id='oper'>  x </button>
                </div>
                <div className='third-row'>
                    <button className='button' data-id='number'>4</button>
                    <button className='button' data-id='number'>5</button>
                    <button className='button' data-id='number'>6</button>
                    <button className='button' data-id='oper'> - </button>
                </div>
                <div className='fourth-row'>
                    <button className='button' data-id='number'>1</button>
                    <button className='button' data-id='number'>2</button>
                    <button className='button' data-id='number'>3</button>
                    <button className='button' data-id='oper'> + </button>
                </div>
                <div className='fifth-row'>
                    <button className='button zero' data-id='number'>0</button>
                    <button className='button' data-id='number'> . </button>
                    <button className='button' data-id='number'> = </button>
                </div>
            </div>
        </div>
    )
}

export default Calculator

