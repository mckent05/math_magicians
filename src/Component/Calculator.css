.result {
  background-color: grey;
  width: 100%;
  height: 90px;
  justify-content: flex-end;
}

.result p {
  font-size: 40px;
  color: white;
}

.btn-cont {
  width: 49vw;
  flex-direction: column;
  margin: 50px auto;
}

.btn-container {
  width: 100%;
}

.button {
  border: 1px solid gainsboro;
  outline: none;
  padding: 12px;
  background-color: rgb(241, 237, 237);
  width: 25%;
  height: 80px;
  font-size: 20px;
  font-weight: 600;
}

.d-flex {
  display: flex;
}

.a-center {
  align-items: center;
}

.j-center {
  justify-content: flex-end;
}

.zero {
  width: 50%;
}

.first-row button:last-of-type,
.second-row button:last-of-type,
.third-row button:last-of-type,
.fourth-row button:last-of-type,
.fifth-row button:last-of-type {
  background-color: darkorange;
}
