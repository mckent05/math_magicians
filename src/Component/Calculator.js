/* eslint-disable react/prop-types, react/prefer-stateless-function */
import React, { Component } from 'react';
import './Calculator.css';

class Calculator extends Component {
  render() {
    const tope = this.props;
    return (
      <div className="btn-cont d-flex j-center a-center">
        <div className="result d-flex a-center">
          <p>
            {tope.result}
          </p>
        </div>
        <div className="btn-container">
          <div className="first-row">
            <button className="button" data-id="AC" type="button">AC</button>
            <button className="button" data-id="oper" type="button">+/-</button>
            <button className="button" data-id="oper" type="button">%</button>
            <button className="button" data-id="oper" type="button"> / </button>
          </div>
          <div className="second-row">
            <button className="button" data-id="number" type="button">7</button>
            <button className="button" data-id="number" type="button">8</button>
            <button className="button" data-id="number" type="button">9</button>
            <button className="button" data-id="oper" type="button">  x </button>
          </div>
          <div className="third-row">
            <button className="button" data-id="number" type="button">4</button>
            <button className="button" data-id="number" type="button">5</button>
            <button className="button" data-id="number" type="button">6</button>
            <button className="button" data-id="oper" type="button"> - </button>
          </div>
          <div className="fourth-row">
            <button className="button" data-id="number" type="button">1</button>
            <button className="button" data-id="number" type="button">2</button>
            <button className="button" data-id="number" type="button">3</button>
            <button className="button" data-id="oper" type="button"> + </button>
          </div>
          <div className="fifth-row">
            <button className="button zero" data-id="number" type="button">0</button>
            <button className="button" data-id="number" type="button"> . </button>
            <button className="button" data-id="number" type="button"> = </button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
