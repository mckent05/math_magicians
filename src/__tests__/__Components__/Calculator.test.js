import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Calculator from '../../Component/Calculator';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing calculator and events', () => {
  it('testing for events on calculator', () => {
    act(() => {
      render(<Calculator />, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();

    
    const event1 = fireEvent.click(screen.getByText('1'));
    expect(event1).toBe(true);

    const event2 = fireEvent.click(screen.getByText('2'));
    expect(event2).toBe(true);

    const event3 = fireEvent.click(screen.getByText('-'));
    expect(event3).toBe(true);

    const event4 = fireEvent.click(screen.getByText('7'));
    expect(event4).toBe(true);
  });
});
