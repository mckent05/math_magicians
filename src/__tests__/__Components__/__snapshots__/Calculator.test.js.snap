// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing calculator and events testing for events on calculator 1`] = `
"<div class=\\"btn-cont d-flex j-center a-center\\">
  <div class=\\"header-title\\">
    <h2>Let's do Some Maths</h2>
  </div>
  <div class=\\"btn-container\\">
    <div class=\\"result d-flex a-center\\">
      <p id=\\"result\\"></p>
    </div>
    <div class=\\"first-row\\"><button class=\\"button\\" data-id=\\"AC\\" type=\\"button\\">AC</button><button class=\\"button\\" data-id=\\"+/-\\" type=\\"button\\">+/-</button><button class=\\"button\\" data-id=\\"%\\" type=\\"button\\">%</button><button class=\\"button\\" data-id=\\"รท\\" type=\\"button\\">รท</button></div>
    <div class=\\"second-row\\"><button class=\\"button\\" data-id=\\"7\\" type=\\"button\\">7</button><button class=\\"button\\" data-id=\\"8\\" type=\\"button\\">8</button><button class=\\"button\\" data-id=\\"9\\" type=\\"button\\">9</button><button class=\\"button\\" data-id=\\"x\\" type=\\"button\\"> x </button></div>
    <div class=\\"third-row\\"><button class=\\"button\\" data-id=\\"4\\" type=\\"button\\">4</button><button class=\\"button\\" data-id=\\"5\\" type=\\"button\\">5</button><button class=\\"button\\" data-id=\\"6\\" type=\\"button\\">6</button><button class=\\"button\\" data-id=\\"-\\" type=\\"button\\">-</button></div>
    <div class=\\"fourth-row\\"><button class=\\"button\\" data-id=\\"1\\" type=\\"button\\">1</button><button class=\\"button\\" data-id=\\"2\\" type=\\"button\\">2</button><button class=\\"button\\" data-id=\\"3\\" type=\\"button\\">3</button><button class=\\"button\\" data-id=\\"+\\" type=\\"button\\">+</button></div>
    <div class=\\"fifth-row\\"><button class=\\"button zero\\" data-id=\\"0\\" type=\\"button\\">0</button><button class=\\"button\\" data-id=\\".\\" type=\\"button\\">.</button><button class=\\"button\\" data-id=\\"=\\" type=\\"button\\">=</button></div>
  </div>
</div>"
`;
